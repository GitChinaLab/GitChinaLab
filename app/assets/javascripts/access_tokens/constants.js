// Token types
export const FEED_TOKEN = 'feedToken';
export const INCOMING_EMAIL_TOKEN = 'incomingEmailToken';
export const STATIC_OBJECT_TOKEN = 'staticObjectToken';
