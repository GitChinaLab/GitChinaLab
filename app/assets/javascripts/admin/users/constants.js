import { s__, __ } from '~/locale';

export const USER_AVATAR_SIZE = 32;

export const LENGTH_OF_USER_NOTE_TOOLTIP = 100;

export const I18N_USER_ACTIONS = {
  edit: __('Edit'),
  userAdministration: s__('AdminUsers|User administration'),
  unlock: __('Unlock'),
  block: s__('AdminUsers|Block'),
  unblock: s__('AdminUsers|Unblock'),
  approve: s__('AdminUsers|Approve'),
  reject: s__('AdminUsers|Reject'),
  deactivate: s__('AdminUsers|Deactivate'),
  activate: s__('AdminUsers|Activate'),
  ldapBlocked: s__('AdminUsers|Cannot unblock LDAP blocked users'),
  delete: s__('AdminUsers|Delete user'),
  deleteWithContributions: s__('AdminUsers|Delete user and contributions'),
  ban: s__('AdminUsers|Ban user'),
  unban: s__('AdminUsers|Unban user'),
};

export const CONFIRM_DELETE_BUTTON_SELECTOR = '.js-delete-user-modal-button';

export const MODAL_TEXTS_CONTAINER_SELECTOR = '#js-modal-texts';

export const MODAL_MANAGER_SELECTOR = '#js-delete-user-modal';
