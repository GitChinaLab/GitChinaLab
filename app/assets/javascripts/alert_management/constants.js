import { s__ } from '~/locale';

export const SEVERITY_LEVELS = {
  CRITICAL: s__('severity|Critical'),
  HIGH: s__('severity|High'),
  MEDIUM: s__('severity|Medium'),
  LOW: s__('severity|Low'),
  INFO: s__('severity|Info'),
  UNKNOWN: s__('severity|Unknown'),
};

export const ALERTS_STATUS_TABS = [
  {
    title: s__('AlertManagement|Open'),
    status: 'OPEN',
    filters: ['TRIGGERED', 'ACKNOWLEDGED'],
  },
  {
    title: s__('AlertManagement|Triggered'),
    status: 'TRIGGERED',
    filters: 'TRIGGERED',
  },
  {
    title: s__('AlertManagement|Acknowledged'),
    status: 'ACKNOWLEDGED',
    filters: 'ACKNOWLEDGED',
  },
  {
    title: s__('AlertManagement|Resolved'),
    status: 'RESOLVED',
    filters: 'RESOLVED',
  },
  {
    title: s__('AlertManagement|All alerts'),
    status: 'ALL',
    filters: ['TRIGGERED', 'ACKNOWLEDGED', 'RESOLVED'],
  },
];

/* eslint-disable @gitlab/require-i18n-strings */

/**
 * Tracks snowplow event when user views alerts list
 */
export const trackAlertListViewsOptions = {
  category: 'Alert Management',
  action: 'view_alerts_list',
};
