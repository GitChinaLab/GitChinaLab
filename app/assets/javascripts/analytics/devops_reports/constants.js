import { __ } from '~/locale';

export const INTRO_COOKIE_KEY = 'dev_ops_report_intro_callout_dismissed';

export const INTRO_BANNER_TITLE = __('Introducing Your DevOps Reports');

export const INTRO_BANNER_BODY = __(
  'Your DevOps Reports give an overview of how you are using GitLab from a feature perspective. Use them to view how you compare with other organizations, and how your teams compare against each other.',
);

export const INTRO_BANNER_ACTION_TEXT = __('Read more');
