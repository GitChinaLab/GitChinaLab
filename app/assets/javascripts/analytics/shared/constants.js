import { masks } from 'dateformat';

export const DATE_RANGE_LIMIT = 180;
export const OFFSET_DATE_BY_ONE = 1;
export const PROJECTS_PER_PAGE = 50;

const { isoDate, mediumDate } = masks;
export const dateFormats = {
  isoDate,
  defaultDate: mediumDate,
  defaultDateTime: 'mmm d, yyyy h:MMtt',
  month: 'mmmm',
};
