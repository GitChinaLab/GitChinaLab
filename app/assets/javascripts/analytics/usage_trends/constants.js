import { getDateInPast } from '~/lib/utils/datetime_utility';

export const TOTAL_DAYS_TO_SHOW = 365;
export const TODAY = new Date();
export const START_DATE = getDateInPast(TODAY, TOTAL_DAYS_TO_SHOW);
