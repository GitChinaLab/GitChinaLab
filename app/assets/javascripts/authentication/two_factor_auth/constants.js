export const COPY_BUTTON_ACTION = 'copy';
export const DOWNLOAD_BUTTON_ACTION = 'download';
export const PRINT_BUTTON_ACTION = 'print';

export const TRACKING_LABEL_PREFIX = '2fa_recovery_codes_';

export const RECOVERY_CODE_DOWNLOAD_FILENAME = 'gitlab-recovery-codes.txt';

export const SUCCESS_QUERY_PARAM = 'two_factor_auth_enabled_successfully';

export const COPY_KEYBOARD_SHORTCUT = 'mod+c';
