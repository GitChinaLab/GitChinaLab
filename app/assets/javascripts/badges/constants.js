export const GROUP_BADGE = 'group';
export const PROJECT_BADGE = 'project';
