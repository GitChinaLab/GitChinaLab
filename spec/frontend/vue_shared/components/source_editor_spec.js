import { shallowMount } from '@vue/test-utils';
import { nextTick } from 'vue';
import { EDITOR_READY_EVENT } from '~/editor/constants';
import Editor from '~/editor/source_editor';
import SourceEditor from '~/vue_shared/components/source_editor.vue';

jest.mock('~/editor/source_editor');

describe('Source Editor component', () => {
  let wrapper;
  let mockInstance;

  const value = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.';
  const fileName = 'lorem.txt';
  const fileGlobalId = 'snippet_777';
  const createInstanceMock = jest.fn().mockImplementation(() => {
    mockInstance = {
      onDidChangeModelContent: jest.fn(),
      updateModelLanguage: jest.fn(),
      getValue: jest.fn(),
      setValue: jest.fn(),
      dispose: jest.fn(),
    };
    return mockInstance;
  });

  Editor.mockImplementation(() => {
    return {
      createInstance: createInstanceMock,
    };
  });
  function createComponent(props = {}) {
    wrapper = shallowMount(SourceEditor, {
      propsData: {
        value,
        fileName,
        fileGlobalId,
        ...props,
      },
    });
  }

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  const triggerChangeContent = (val) => {
    mockInstance.getValue.mockReturnValue(val);
    const [cb] = mockInstance.onDidChangeModelContent.mock.calls[0];

    cb();

    jest.runOnlyPendingTimers();
  };

  describe('rendering', () => {
    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders content', () => {
      expect(wrapper.text()).toContain(value);
    });
  });

  describe('functionality', () => {
    it('does not fail without content', () => {
      const spy = jest.spyOn(global.console, 'error');
      createComponent({ value: undefined });

      expect(spy).not.toHaveBeenCalled();
      expect(wrapper.find('[id^="source-editor-"]').exists()).toBe(true);
    });

    it('initialises Source Editor instance', () => {
      const el = wrapper.find({ ref: 'editor' }).element;
      expect(createInstanceMock).toHaveBeenCalledWith({
        el,
        blobPath: fileName,
        blobGlobalId: fileGlobalId,
        blobContent: value,
        extensions: null,
      });
    });

    it('reacts to the changes in fileName', () => {
      const newFileName = 'ipsum.txt';

      wrapper.setProps({
        fileName: newFileName,
      });

      return nextTick().then(() => {
        expect(mockInstance.updateModelLanguage).toHaveBeenCalledWith(newFileName);
      });
    });

    it('registers callback with editor onChangeContent', () => {
      expect(mockInstance.onDidChangeModelContent).toHaveBeenCalledWith(expect.any(Function));
    });

    it('emits input event when the blob content is changed', () => {
      expect(wrapper.emitted().input).toBeUndefined();

      triggerChangeContent(value);

      expect(wrapper.emitted().input).toEqual([[value]]);
    });

    it('emits EDITOR_READY_EVENT event when the Source Editor is ready', async () => {
      const el = wrapper.find({ ref: 'editor' }).element;
      expect(wrapper.emitted()[EDITOR_READY_EVENT]).toBeUndefined();

      await el.dispatchEvent(new Event(EDITOR_READY_EVENT));

      expect(wrapper.emitted()[EDITOR_READY_EVENT]).toBeDefined();
    });

    it('component API `getEditor()` returns the editor instance', () => {
      expect(wrapper.vm.getEditor()).toBe(mockInstance);
    });

    describe('reaction to the value update', () => {
      it('reacts to the changes in the passed value', async () => {
        const newValue = 'New Value';

        wrapper.setProps({
          value: newValue,
        });

        await nextTick();
        expect(mockInstance.setValue).toHaveBeenCalledWith(newValue);
      });

      it("does not update value if the passed one is exactly the same as the editor's content", async () => {
        const newValue = `${value}`; // to make sure we're creating a new String with the same content and not just a reference

        wrapper.setProps({
          value: newValue,
        });

        await nextTick();
        expect(mockInstance.setValue).not.toHaveBeenCalled();
      });
    });
  });
});
