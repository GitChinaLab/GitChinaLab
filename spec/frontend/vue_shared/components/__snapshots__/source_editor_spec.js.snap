// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Source Editor component rendering matches the snapshot 1`] = `
<div
  data-editor-loading=""
  data-qa-selector="source_editor_container"
  id="source-editor-snippet_777"
>
  <pre
    class="editor-loading-content"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </pre>
</div>
`;
